cns/logger/zaplogger.go

```diff
@@ -0,0 +1,18 @@
+package logger
+
+import (
+	"os"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+func initZapLogger(loggingLevel zapcore.Level, encoder zapcore.Encoder) (*zap.Logger, error) {
+	platformCore, err := GetPlatformCores(loggingLevel, encoder)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get Platform cores")
+	}
+	// Create a new logger with the platform core.
+	return zap.New(platformCore, zap.AddCaller()).With(zap.Int("pid", os.Getpid())), nil
+}
```

Comment:
Seems like we could do this in `NewCNSLogger()` also and don't need it extracted in to its own func